packages/protocol/src/abstracts/BaseRouter.sol

```diff
@@ -62,30 +60,22 @@ abstract contract BaseRouter is SystemAccessControl, IRouter {
 
   /**
    * @dev Stores token balances of this contract at a given moment.
-   * It's used to ensure there're no changes in balances at the
-   * end of a transaction.
+   * It's used to pass tokens to check from parent contract this contract.
    */
-  Snapshot[] internal _tokensToCheck;
-
-  /**
-   * @dev Operations in the bundle should "benefit" or be executed
-   * on behalf of this account. These are receivers on DEPOSIT and PAYBACK
-   * or owners on WITHDRAW and BORROW.
-   */
-  address private _beneficiary;
+  Snapshot internal _tempTokenToCheck;
 
   /**
    * @notice Constructor of a new {BaseRouter}.
    *
    * @param weth wrapped native token of this chain
    * @param chief contract
    */
-  constructor(IWETH9 weth, IChief chief) SystemAccessControl(address(chief)) {
```

Comment:
why does the constructor need to be "payable"?

Reply:
Payable functions save gas [reference](https://dev.to/zaryab2000/does-adding-a-payable-keyword-in-solidity-actually-save-gas-24bm). I implemented some subtle suggestions from the Hats findings.